### ISSUE TITLE
how to combine deepspeed with megatron

### ISSUE THREAD

**Issue by Stick-To** [microsoft/DeepSpeed#3502]
when I train with this code, the loss become very large after small number of steps
```
deepspeed.init_distributed("nccl")
mpu.initialize_model_parallel(int(args.tp), int(args.pp))
mpu.model_parallel_cuda_manual_seed(1234)
mpu.checkpoint = deepspeed.checkpointing.checkpoint
mpu.get_cuda_rng_tracker = deepspeed.checkpointing.get_cuda_rng_tracker
mpu.model_parallel_cuda_manual_seed = deepspeed.checkpointing.model_parallel_cuda_manual_seed
topo = PipeModelDataParallelTopology(num_pp=mpu.get_pipeline_model_parallel_world_size(),
                                     num_mp=mpu.get_tensor_model_parallel_world_size(),
                                     num_dp=mpu.get_data_parallel_world_size())
model = PPLlamaForCausalLMSpec(
                                    partition_method="uniform",
                                    activation_checkpoint_interval=1,
                                    checkpointable_layers=["LlamaDecoderLayer"],
                                    topology=topo
                                   )
engine, optim, _, scheduler = deepspeed.initialize(
        model=model,
        model_parameters=model.parameters(),
        training_data=dataset,
        optimizer=optim,
        lr_scheduler=lrscheduler,
        config=deepspeed_config,
    )

for step in range(num_iters):
    loss = engine.train_batch()
```
```
Using topology: {ProcessCoord(pipe=0, data=0, model=0): 0, ProcessCoord(pipe=0, data=0, model=1): 1, ProcessCoord(pipe=1, data=0, model=0): 2, ProcessCoord(pipe=1, data=0, model=1): 3, ProcessCoord(pipe=2, data=0, model=0): 4, ProcessCoord(pipe=2, data=0, model=1): 5, ProcessCoord(pipe=3, data=0, model=0): 6, ProcessCoord(pipe=3, data=0, model=1): 7}
```

```
steps: 1 loss: 1.5171 iter time (s): 35.339 samples/sec: 0.453
[2023-05-10 16:33:11,501] [INFO] [logging.py:96:log_dist] [Rank 0] step=2, skipped=0, lr=[4.110996916752313e-08], mom=[(0.9, 0.999)]
steps: 2 loss: 10.5195 iter time (s): 28.429 samples/sec: 0.563
[2023-05-10 16:33:39,455] [INFO] [logging.py:96:log_dist] [Rank 0] step=3, skipped=0, lr=[6.166495375128469e-08], mom=[(0.9, 0.999)]
steps: 3 loss: 10.5238 iter time (s): 27.975 samples/sec: 0.572
```
